client/src/templates/Challenges/components/project-preview-modal.tsx

```diff
@@ -0,0 +1,104 @@
+import { Button, Modal } from '@freecodecamp/react-bootstrap';
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { connect } from 'react-redux';
+
+import type { ChallengeFile, Required } from '../../../redux/prop-types';
+import {
+  closeModal,
+  setEditorFocusability,
+  isProjectPreviewModalOpenSelector,
+  projectPreviewMounted
+} from '../redux';
+import { projectPreviewId } from '../utils/frame';
+import Preview from './preview';
+
+import './project-preview-modal.css';
+
+export interface PreviewConfig {
+  challengeType: boolean;
+  challengeFiles: ChallengeFile[];
+  required: Required;
+  template: string;
+}
+
+interface Props {
+  closeModal: () => void;
+  isOpen: boolean;
+  projectPreviewMounted: (previewConfig: PreviewConfig) => void;
+  previewConfig: PreviewConfig;
+  setEditorFocusability: (focusability: boolean) => void;
+}
+
+const mapStateToProps = (state: unknown) => ({
+  isOpen: isProjectPreviewModalOpenSelector(state) as boolean
+});
+const mapDispatchToProps = {
+  closeModal: () => closeModal('projectPreview'),
```

Comment:
I am not a fan of this reassignment (same name, different function). We may do it elsewhere, but I would prefer to be able to globally search `closeModal` and the definition and usage be: `(modalName: string) => void`

```suggestion
  closeModal,
```